nestjs/src/courses/course-students/course-students.service.ts

```diff
@@ -0,0 +1,179 @@
+    const record = await this.mentorRepository.findOne({
+      relations: ['user'],
+      where: {
+        id: mentorId,
+      },
+    });
+
+    if (!record) {
+      throw new NotFoundException(`Mentor not found ${mentorId}`);
+    }
+
+    const mentor = convertToMentorBasic(record);
+    const user = record.user as User;
+    const mentorWithContacts: MentorWithContacts = {
+      ...mentor,
+      contactsEmail: user.contactsEmail,
+      contactsSkype: user.contactsSkype,
+      contactsWhatsApp: user.contactsWhatsApp,
+      contactsTelegram: user.contactsTelegram,
+      contactsNotes: user.contactsNotes,
+      contactsPhone: null,
+    };
+    return mentorWithContacts;
+  }
+}
+
+const getStageInterviewRating = (stageInterviews: StageInterview[]) => {
+  const [lastInterview] = stageInterviews
+    .filter((interview: StageInterview) => interview.isCompleted)
+    .map(({ stageInterviewFeedbacks, score }: StageInterview) =>
+      stageInterviewFeedbacks.map((feedback: StageInterviewFeedback) => ({
+        date: feedback.updatedDate,
+        // interviews in new template should have score precalculated
+        rating: score ?? getInterviewRatings(JSON.parse(feedback.json) as StageInterviewFeedbackJson).rating,
+      })),
+    )
+    .reduce((acc, cur) => acc.concat(cur), [])
+    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime());
+
+  return lastInterview && lastInterview.rating !== undefined ? lastInterview.rating : null;
+};
+
+export function getInterviewRatings({ skills, programmingTask, resume }: StageInterviewFeedbackJson) {
+  const commonSkills = Object.values(skills?.common ?? {}).filter(Boolean) as number[];
+  const dataStructuresSkills = Object.values(skills?.dataStructures ?? {}).filter(Boolean) as number[];
+
+  const htmlCss = skills?.htmlCss.level;
+  const common = commonSkills.reduce((acc, cur) => acc + cur, 0) / commonSkills.length;
+  const dataStructures = dataStructuresSkills.reduce((acc, cur) => acc + cur, 0) / dataStructuresSkills.length;
+
+  if (resume?.score !== undefined) {
+    const rating = resume.score;
+    return { rating, htmlCss, common, dataStructures };
+  }
+
+  const ratingsCount = 4;
+  const ratings = [htmlCss, common, dataStructures, programmingTask.codeWritingLevel].filter(Boolean) as number[];
+  const rating = (ratings.length === ratingsCount ? ratings.reduce((sum, num) => sum + num) / ratingsCount : 0) * 10;
+
+  return { rating, htmlCss, common, dataStructures };
+}
+
+export function convertToMentorBasic(mentor: Mentor): MentorBasic {
+  const user = (mentor.user as User)!;
+  return {
+    isActive: !mentor.isExpelled,
+    name: createName(user),
+    id: mentor.id,
+    githubId: user.githubId,
+    students: mentor.students ? mentor.students.filter(s => !s.isExpelled && !s.isFailed).map(s => ({ id: s.id })) : [],
+    cityName: user.cityName ?? '',
+    countryName: user.countryName ?? '',
+  };
```

Comment:
Type of mentor.user is User without this typecast
```suggestion
  return {
    isActive: !mentor.isExpelled,
    name: createName(mentor.user),
    id: mentor.id,
    githubId: user.githubId,
    students: mentor.students ? mentor.students.filter(s => !s.isExpelled && !s.isFailed).map(s => ({ id: s.id })) : [],
    cityName: user.cityName ?? '',
    countryName: user.countryName ?? '',
  };
```